src/rust/catloop/queue.rs

```diff
@@ -88,6 +88,9 @@ impl CatloopQueue {
     where
         F: FnOnce(Yielder) -> Result<TaskHandle, Fail>,
     {
+        // Check whether we are accepting on this queue.
+        self.socket.borrow_mut().prepare_accept()?;
```

Comment:
Is there a reason to do this here instead of inside the socket? I'd like the queue to not have to interact with the socket state.